src/SupportedEntities.php

```diff
@@ -0,0 +1,1459 @@
+<?php
+
+namespace Drupal\civicrm_entity;
+
+/**
+ * Defines supported entities.
+ *
+ * See `$civicrm_entity_info['civicrm_event']` for implemented usage. This might
+ * not be required. Perhaps we just implement entity class definitions using
+ * this information. Providing entity type derivatives seems hard, and possibly
+ * overhead when we just can provide the classes directly.
+ *
+ * Entities could just define the `civicrm_entity` key to define their support
+ * and what CiviCRM entity they map to.
+ *
+ * Ported for now and used in civicrm_entity_entity_type_build()
+ *
+ * @see civicrm_entity_entity_type_build()
+ */
+final class SupportedEntities {
+
+  public static function getInfo() {
+    $civicrm_entity_info = [];
+    $civicrm_entity_info['civicrm_activity'] = [
+      'civicrm entity name' => 'activity',
+      'label property' => 'subject',
+      'permissions' => [
+        'view' => ['view all activities'],
+        'edit' => [],
+        'update' => [],
+        'create' => [],
+        'delete' => ['delete activities'],
+      ],
+      'theme' => [
+        'template' => 'civicrm-activity',
+        'path' => drupal_get_path('module', 'civicrm_entity') . '/templates',
+      ],
+      'display suite' => [
```

Comment:
I'm wondering if the display suite section of the supported entities metadata will be necessary now.  I wonder if we aim to support the new Layout experimental modules in 8.4 and 8.5.  Not an expert on the nuances of display suite vs. the new Layout stuff going into Core soon. @mglaman your thoughts on this?  Do we need Display Suite anymore to accomplish what we did with the D7 version.....I used Display suite integration in the D7 version so that entity properties could have display formatters. Is this necessary now?

Reply:
IMO, it's hard to say what will win out at this point. Layout Discovery (new core module) is for sure a thing: it's used by Display Suite, Panels and Field Layout (the core DS-like thing). Why not support them all? ;-)

I don't know DS or Field Layout that well, so I'm not sure what's necessary to integrate with them. But if we do the Entity/Field API right, then Panels/Panelizer should mostly just work (a plugin is necessary to say the entity type supports Panelizer - I can help with that bit. :-))

Reply:
Definitely want to support the Core Layout Discovery, but of course been loving,knowing.loving  Display Suite in D7,...I know you'lls interest in this is maybe the REST API / Services side of things, but I've spent a a ton of time, and have a lot of clients buying into front side things, like I've replicated the civicrm event registration process in Drupal 7 FAPI , and I want to do the contribution form too for the D8 version. This way us Drupal folk can have Drupal FAPI forms we can customize our way :) The way it sounds with Layout discovery and all those modules using it, we can have several cakes and eat them too. I'm excited to be able to start porting my D7 FAPI custom field types to D8

Reply:
I got an eye toward some some Commerce integration too...